### PR TITLE
Eml-to-pdf bug fixes: removal of incompatible fonts, removal of emoji in favor of @, jakarta-mail dependency handling improvements

### PR DESCRIPTION
**Pull request by balazs-szucs**
# Description of Changes
This pull request introduces enhancements and code cleanup to the `EmlToPdf` utility class, focusing on improving email-to-PDF conversion, handling embedded images, and simplifying the codebase. Key changes include better handling of inline images, enhanced Jakarta Mail dependency checks, and refactoring for improved readability and maintainability.

### Enhancements to Email-to-PDF Conversion:
* Added support for processing inline images (`cid:` references) by converting them into data URIs for proper inline display. 
* Improved attachment handling to always include embedded images regardless of size, ensuring inline display functionality. 
* Enhanced email HTML generation to process inline images and include them in the email body.

### Attachment Handling Enhancements:
* Replaced the attachment icon placeholder (`icon` or 📎 emoji) with a new marker (`@`) for consistency across the application (non-fat images did not support the emoji, however @ is supported accross the board.)
* Updated the annotation logic to use `AttachmentMarkerPositionFinder` instead of `EmojiPositionFinder`, aligning with the new attachment marker system.

### Jakarta Mail Dependency Handling:
* Added detailed checks for core Jakarta Mail classes to determine availability in different environments (e.g., Docker). 
* Introduced validation for Jakarta Mail multipart and part types to prevent processing invalid objects.
* Explicitly parse in the classes:

- jakarta.mail.internet.MimeMessage – Core email message parsing
- jakarta.mail.Session – Email session management
- jakarta.mail.internet.MimeUtility – MIME encoding/decoding utilities
- jakarta.mail.internet.MimePart – Individual MIME parts (attachments, body parts)
- jakarta.mail.internet.MimeMultipart – Multi-part MIME messages
- jakarta.mail.Multipart – Base multipart interface
- jakarta.mail.Part – Base part interface

### Code Cleanup and Refactoring:
* Simplified utility classes (`StyleConstants`, `MimeConstants`, `FileSizeConstants`) by removing unnecessary constructors and unused constants.
* Updated log messages for clarity, such as distinguishing between general content processing errors and multipart-specific issues.
---

### General

- [x] I have read the [Contribution Guidelines](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/CONTRIBUTING.md)
- [x] I have read the [Stirling-PDF Developer Guide](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/DeveloperGuide.md) (if applicable)
- [ ] I have read the [How to add new languages to Stirling-PDF](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/HowToAddNewLanguage.md) (if applicable)
- [x] I have performed a self-review of my own code
- [ ] My changes generate no new warnings

### Documentation

- [ ] I have updated relevant docs on [Stirling-PDF's doc repo](https://github.com/Stirling-Tools/Stirling-Tools.github.io/blob/main/docs/) (if functionality has heavily changed)
- [ ] I have read the section [Add New Translation Tags](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/HowToAddNewLanguage.md#add-new-translation-tags) (for new translation tags only)

### UI Changes (if applicable)

- [ ] Screenshots or videos demonstrating the UI changes are attached (e.g., as comments or direct attachments in the PR)

### Testing (if applicable)

- [x] I have tested my changes locally. Refer to the [Testing Guide](https://github.com/Stirling-Tools/Stirling-PDF/blob/main/DeveloperGuide.md#6-testing) for more details.

### CODE REVIEW

File: common/src/main/java/stirling/software/common/util/EmlToPdf.java [hunk 1]
```diff
@@ -104,13 +95,30 @@ private FileSizeConstants() {
...
+                Class.forName("jakarta.mail.Part");
+
                 jakartaMailAvailable = true;
                 log.debug("Jakarta Mail libraries are available");
             } catch (ClassNotFoundException e) {
-                jakartaMailAvailable = false;
-                log.debug("Jakarta Mail libraries are not available, using basic parsing");
+                // In Docker/production environments, if we detect we're in a containerized
+                // environment
+                String dockerEnv = System.getenv("DOCKER_ENABLE_SECURITY");
```
Comment (Frooodle) [inline common/src/main/java/stirling/software/common/util/EmlToPdf.java]: Better to check using bean (or local similar to)
 @Bean(name = "RunningInDocker")
    public boolean runningInDocker() {
        return Files.exists(Paths.get("/.dockerenv"));
    }
 @Bean(name = "machineType")
    public String determineMachineType() {
        try {
            boolean isDocker = runningInDocker();
            boolean isKubernetes = System.getenv("KUBERNETES_SERVICE_HOST") != null;
            boolean isBrowserOpen = "true".equalsIgnoreCase(env.getProperty("BROWSER_OPEN"));

            if (isKubernetes) {
                return "Kubernetes";
            } else if (isDocker) {
                return "Docker";
            } else if (isBrowserOpen) {
                String os = System.getProperty("os.name").toLowerCase(Locale.ROOT);
                if (os.contains("win")) {
                    return "Client-windows";
                } else if (os.contains("mac")) {
                    return "Client-mac";
                } else {
                    return "Client-unix";
                }
            } else {
                return "Server-jar";
            }
        } catch (Exception e) {
            return "Unknown";
        }
    }

Comment (Frooodle) [reply common/src/main/java/stirling/software/common/util/EmlToPdf.java]: Although dont think there is much need to check anyway looking at feature?

**Comment by Frooodle** [comment]
what is it missing that causes the Jakarta Mail libraries to be missing? what system dep?

File: common/src/main/java/stirling/software/common/util/EmlToPdf.java (hunk 1 above)
Comment (balazs-szucs) [reply common/src/main/java/stirling/software/common/util/EmlToPdf.java]: Thanks for the suggestion. I think I agree with 2nd better that it is not necessary in prod, but was useful while in development. Addressed in [f301545](https://github.com/Stirling-Tools/Stirling-PDF/pull/3770/commits/f301545ad3ec9944c8f8b293068446f56bc0bd66)

**Comment by balazs-szucs** [comment]
I do think that the font was my own fault, but Jakarta Mail problem was partly missing version.properties. I added better fallback to this, and kind more explicit checks. I tried to be more explicit in how the especially the content part handled e.g in processMultipartAdvanced/Image handling just in case and that seem to solve the other part of it. In my testing, when I added logging there it seemed like it wasn't even called or reached by the code when in docker. Not 100% sure why. 

Anyhow, I spent considerable time testing this with all sorts of edge cases, and I am confident about this.

**Comment by balazs-szucs** [comment]
Do you think it would be wise to add junit Test to this? I think that would levitate some concerns , and making emails for testing that would fail without proper dependencies wouldn't be that hard. But I didn't check this beforehand so this is just a guess.

**Comment by Frooodle** [comment]
JUnits are welcome to add but not a requirement for the repo, although we should start requiring that tbf...

**Comment by balazs-szucs** [comment]
Makes sense. I wouldn't mix Junit test with this per se, but I'll work on that then. Sorry for inconvenience on this.

**Comment by Frooodle** [comment]
Ill merge this code as is, junits can be separate

**Review by Frooodle — APPROVED** [review]
